quickwit-indexing/src/merge_policy.rs

```diff
@@ -260,29 +260,30 @@ impl StableMultitenantWithTimestampMergePolicy {
     ///   protect against too big splits as it will break the building of demux operations, see
     ///   [`build_first_demux_operation`].
     fn is_mature_for_demux(&self, split: &SplitMetadata) -> bool {
-        // If demux is disabled, split is considered mature as it will not undergo a demux
-        // operation.
-        if !self.demux_enabled {
+        // If demux is disabled or if there is no demux field, split is considered mature.
+        if !self.demux_enabled || self.demux_field_name.is_none() {
             return true;
         }
-        let demux_field_name = if let Some(demux_field_name) = self.demux_field_name.as_ref() {
-            demux_field_name
-        } else {
-            // All splits are considered mature if there is no demux field as no
-            // split will be demuxed.
-            return true;
-        };
         if split.num_docs >= self.demux_factor * self.split_num_docs_target {
             return true;
         }
-        let split_tags_contains_less_than_2_demux_values = split
-            .tags
-            .iter()
-            .filter(|tag| match_tag_field_name(demux_field_name, tag))
-            .count()
-            < 2;
-        split_tags_contains_less_than_2_demux_values
-            || (split.num_docs < self.split_num_docs_target || split.demux_num_ops > 0)
+        let demux_field_name = self.demux_field_name.as_ref().unwrap();
+        // Note: split metadata tags is empty when demux values have a cardinality
+        // is strictly superior to [`MAX_VALUES_PER_TAG_FIELD`]. When inferior,
+        // tags always contains the special tag `field_tag(demux_field_name)`.
+        // This is why we have to check first if this special tag is present and then
+        // consider the split mature if we have less that one demux value.
```

Comment:
```suggestion
        // consider the split mature if we have less than one demux value.
```

Reply:
I still need to count() all tags like `field_name:value`. The count can be 0 or 1.